experimental/fluent/agentic/src/test/java/io/serverlessworkflow/fluent/agentic/LC4JEquivalenceIT.java

```diff
@@ -0,0 +1,288 @@
+    var a2 = AgentsUtils.newAudienceEditor();
+    var a3 = AgentsUtils.newStyleEditor();
+
+    Workflow wf = workflow("seqFlow").tasks(tasks -> tasks.sequence("process", a1, a2, a3)).build();
+
+    List<TaskItem> items = wf.getDo();
+    assertThat(items).hasSize(3);
+
+    assertThat(items.get(0).getName()).isEqualTo("process-0");
+    assertThat(items.get(1).getName()).isEqualTo("process-1");
+    assertThat(items.get(2).getName()).isEqualTo("process-2");
+    items.forEach(it -> assertThat(it.getTask().getCallTask()).isInstanceOf(CallTaskJava.class));
+
+    Map<String, Object> input =
+        Map.of(
+            "topic", "dragons and wizards",
+            "style", "fantasy",
+            "audience", "young adults");
+
+    Map<String, Object> result;
+    try (WorkflowApplication app = WorkflowApplication.builder().build()) {
+      result = app.workflowDefinition(wf).instance(input).start().get().asMap().orElseThrow();
+    } catch (Exception e) {
+      throw new RuntimeException("Workflow execution failed", e);
+    }
+
+    assertThat(result).containsKey("story");
+  }
+
+  @Test
+  @DisplayName("Looping agents via DSL.loop(...)")
+  public void loopWorkflow() {
+
+    var scorer = AgentsUtils.newStyleScorer();
+    var editor = AgentsUtils.newStyleEditor();
+
+    Workflow wf =
+        AgentWorkflowBuilder.workflow("retryFlow")
+            .loop("reviewLoop", c -> c.readState("score", 0).doubleValue() >= 0.8, scorer, editor)
+            .build();
+
+    List<TaskItem> items = wf.getDo();
+    assertThat(items).hasSize(1);
+
+    var fn = (ForTaskFunction) items.get(0).getTask().getForTask();
+    assertThat(fn.getDo()).isNotNull();
+    assertThat(fn.getDo()).hasSize(2);
+    fn.getDo()
+        .forEach(si -> assertThat(si.getTask().getCallTask()).isInstanceOf(CallTaskJava.class));
+
+    Map<String, Object> input =
+        Map.of(
+            "story", "dragons and wizards",
+            "style", "comedy");
+
+    Map<String, Object> result;
+    try (WorkflowApplication app = WorkflowApplication.builder().build()) {
+      result = app.workflowDefinition(wf).instance(input).start().get().asMap().orElseThrow();
+    } catch (Exception e) {
+      throw new RuntimeException("Workflow execution failed", e);
+    }
+
+    assertThat(result).containsKey("story");
+  }
+
+  @Test
+  @DisplayName("Looping agents via DSL.loop(...)")
+  public void loopWorkflowWithMaxIterations() {
+    var scorer = AgentsUtils.newStyleScorer();
+    var editor = AgentsUtils.newStyleEditor();
+
+    Workflow wf =
+        AgentWorkflowBuilder.workflow("maxFlow")
+            .tasks(
+                d ->
+                    d.loop(
+                        "limit",
+                        l ->
+                            l.maxIterations(5)
+                                .exitCondition(c -> c.readState("score", 0).doubleValue() >= 0.8)
+                                .subAgents("sub", scorer, editor)))
```

Comment:
Here we should use the shortcuts via `DSL.loop`. You may send another parallel PR to include `maxIterations` to the shortcuts API if needed.